src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joined = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this.length === 1) {
+        joined += this[i];
+        break;
+      }
```

Comment:
this one is redundant, no need to do it (there's gonna be just 1 iteration in such cases, we won't go forward)